src/java/org/apache/cassandra/io/sstable/format/SSTableReader.java

```diff
@@ -1691,7 +1691,7 @@ public boolean markObsolete(Tracker tracker)
         {
             assert !tidy.isReplaced;
         }
-        if (!tidy.global.isCompacted.getAndSet(true))
+        if (!tidy.global.setCompacted())
```

Comment:
note during review:
- this is the only place in the code `isCompacted` was getting set to true
- `setCompacted` is synchronized

thus the former atomic call to getAndSet is still mirrored in this change